w2/internal/service/product.go

```diff
@@ -69,44 +68,77 @@ func (s *ProductService) CreateProduct(req CreateProductRequest) (CreateProductR
 	return resp, nil
 }
 
-func (s *ProductService) UpdateProduct(req UpdateProductRequest) (entity.Product, error) {
-	_, err := s.productRepository.GetById(req.ID)
+func (s *ProductService) UpdateProduct(req UpdateProductRequest, userId string) (entity.Product, commons.CustomError) {
+	productInfo, err := s.productRepository.GetById(req.ID)
 	if err != nil {
-		return entity.Product{}, err
+		return entity.Product{}, commons.CustomError{
+			Message: err.Error(),
+			Code:    500,
+		}
```

Comment:
1. There is a interesting error that returned by GetById method, the error is CustomError with code 404 for handling sql.ErrNoRows error. So, if GetById returns an error, so just return that error again.